File: examples/python/greeter.py
#!/usr/bin/python

# Copyright 2013 Joe Walnes and the websocketd team.
# All rights reserved.
# Use of this source code is governed by a BSD-style
# license that can be found in the LICENSE file.

from sys import stdin, stdout

# For each line FOO received on STDIN, respond with "Hello FOO!".
while True:
  line = stdin.readline().strip()
  print('Hello %s!' % line)
  stdout.flush() # Remember to flush

File: examples/rust/greeter.rs
use std::io::{self, Write};

// For each line FOO received on STDIN, respond with "Hello FOO!".
fn main() {
  loop {
    let mut msg = String::new();
    io::stdin()
      .read_line(&mut msg)
      .expect("Failed to read line");
    let msg = msg.trim();
    println!("Hello {}!", msg);
    io::stdout().flush().ok().expect("Could not flush stdout");
  }
}
